ckanext/cfpb_extrafields/plugin.py

```diff
@@ -76,9 +76,16 @@ def _redirect_to_edit_on_change(self, resource, field):
     def _delete_and_rebuild_datadict(self, resource):
         import json
         if 'datadict' in resource and 'id' in resource:
-            record = resource['datadict']
-            resource.pop('datadict')
-            json_record = json.loads(record)
+            record = resource.pop('datadict')
+
+            try:
+                json_record = json.loads(record)
+            except ValueError as e:
+                # Invalid JSON, so don't remove old data
+                error_message = "Error saving data dictionary: {0}.  Data was: {1}".format(e, record)
+                flash_error(error_message)
```

Comment:
Sounds good. Check out the logs. Ultimately it seems to occur when CKAN attempts some text conversion before sending the data to the Datastore. It chokes on special characters that result from things like new line formatting, and it also seems to happen when there's too much text, which, for some reason, results in what might be a warning that includes special characters. Lots of implied question marks in the above. Happy hunting :)